docs/build/guides/conventions/analyzing-smart-contract-cost.mdx

```diff
@@ -0,0 +1,449 @@
+use serde::{Deserialize, Serialize};
+use regex::Regex;
+
+#[contract]
+pub struct ExampleContract;
+
+#[contractimpl]
+impl ExampleContract {
+    // Function to log a message
+    pub fn log_message(env: Env, message: Symbol) {
+        log!(&env, "Message: {}", message);
+
+        // Unnecessary use of serde and regex
+        let _data: Vec<u8> = serde_json::to_vec(&message).unwrap();
+        let _re = Regex::new(r"\w+").unwrap();
+    }
+}
+```
+
+:::danger
+
+- Unnecessary Libraries: The contract includes serde, serde_json, and regex, which are not essential for the contract's primary functionality.
+- Increased WASM Size: Including these libraries adds unnecessary bloat, increasing the size of the compiled WASM file
+
+:::
+
+#### Example (Optimized Contract) ✅
+
+```rust
+#![no_std]
+use soroban_sdk::{contract, contractimpl, log, Env, Symbol};
+
+#[contract]
+pub struct ExampleContract;
+
+#[contractimpl]
+impl ExampleContract {
+    // Function to log a message
+    pub fn log_message(env: Env, message: Symbol) {
+        log!(&env, "Message: {}", message);
+    }
+}
+```
+
+:::tip
+
+- Essential Libraries Only: The contract only includes soroban_sdk, which is necessary for its functionality.
+- Reduced WASM Size: By avoiding unnecessary libraries, the size of the compiled WASM file is minimized.
+
+:::
+
+## Other tips to optimize smart contract cost:
+
+### 1. **Use of built-in tools**
+
+One way to optimize the size of your `wasm` binary is to use the `soroban optimize` command. This command comes from the [Soroban CLI](https://www.google.com) will optimize the size of your `wasm` binary by removing unnecessary code and reducing the size of the binary.
+
+```bash
+stellar contract optimize \
+  --wasm target/wasm32-unknown-unknown/release/project.wasm
+```
+
+Another way to optimize your smart contract is to use the `stellar contract invoke` command with the `--cost` and `--sim-only` flags. This command will provide you with a detailed breakdown of the cost of running your smart contract on the Stellar network.
+
+```bash
+stellar contract invoke \
+  --id CC6MWZMG2JPQEENRL7XVICAY5RNMHJ2OORMUHXKRDID6MNGXSSOJZLLF \
+  --source alice \
+  --network testnet \
+  --cost \
+  --sim-only \
+  -- \
+  increment
+```
+
+Reference: https://sorobandev.com/guides/soroban-cli
+
+### 2. **Manual code review**
+
+Perform a manual code review of your smart contract to identify areas where you can optimize the code. Look for redundant loops, conditions, and storage operations that can be minimized or removed.
+
+### 3. **Unit testing with gas measurements:**
+
+Use unit tests to measure the gas cost of your smart contract functions. This will help you identify functions that are consuming a lot of gas and optimize them accordingly.
+
+Frameworks like `cargo-contract` provide tools to measure the gas cost of your smart contract functions. Use these tools to identify bottlenecks in your smart contract code.
```

Comment:
Can you cross-reference this? You should probably also mention [simulateTransaction](https://developers.stellar.org/docs/data/rpc/api-reference/methods/simulateTransaction)